### DISCUSSION TITLE
Transitive Dependency support for *Packages.props

### DISCUSSION THREAD

**Discussion by lvanbuiten** [aquasecurity/trivy#6392] [Development]
Is there any chance that Transitive Dependency support for *Packages.props will be added. In contrast to what the Release Note of v0.48.0 suggests (#5588), it currently isn't supported. Therefore we still have to [generate packages.lock.json](https://aquasecurity.github.io/trivy/v0.50/docs/coverage/language/dotnet/#packageslockjson) files for our large repo containing 250+ projects and duplicate detection.

**Comment by DmitriyLewen** [comment]
Hello @lvanbuiten 
If I remember correctly, `*packages.props` files only contain direct dependencies.

Do you know any way/files to get a list of child dependencies for direct dependencies?

**Reply by lvanbuiten** [reply]
Hi @DmitriyLewen 
They do only contain the direct dependencies.

I unfortunately haven't figured it out yet and was hoping someone would know it (@yuriShafet perhaps?). NuGet and other tooling do know the transitive dependencies so it would guess it isn't magic.

After project restoring/building, which also is needed with [generate packages.lock.json](https://aquasecurity.github.io/trivy/v0.50/docs/coverage/language/dotnet/#packageslockjson), there is a file in the obj folder named `project.assets.json` which might lead to a possibility.
Another possibility is to use `dotnet list package --include-transitive` which will generate output for given solution or project.
Both require knowledge of the structure of the targeted repository, which don't look feasible to me.

**Reply by DmitriyLewen** [reply]
Nuget has many controversial and inaccurate points regarding dependencies.

We can try to find the Nuget cache and iterate through it to find indirect dependencies (I’m not sure if this is possible, but we have already used this approach for other language files).
But there may also be unobvious problems here.

I think the best option is to use a lock file. In this case, Nuget itself will resolve conflicts, select the correct indirect dependencies, etc.

**Reply by lvanbuiten** [reply]
We have that option with lock files already in place, so that's definitely can be our answer. I don't think it is worth the effort to implement it either, but was curious if it was on any roadmap which i hadn't found.

Thank you for your time 👍
